pkg/collector/adapters/config_to_probe_test.go

```diff
@@ -0,0 +1,243 @@
+package adapters_test
+
+import (
+	"github.com/open-telemetry/opentelemetry-operator/pkg/collector/adapters"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestSimpleCase(t *testing.T) {
+	configStr := `extensions:
+  health_check:
+service:
+  extensions: [health_check]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	actualProbe, err := adapters.ConfigToContainerProbe(logger, config)
+	assert.NoError(t, err)
+	assert.Equal(t, "/", actualProbe.HTTPGet.Path)
+	assert.Equal(t, int32(13133), actualProbe.HTTPGet.Port.IntVal)
+	assert.Equal(t, "0.0.0.0", actualProbe.HTTPGet.Host)
```

Comment:
This is now interesting: `0.0.0.0`, from the server perspective, means "bind to all IPs". IIRC, it's not a valid IP from the client's perspective. I just tried locally, and `ping 0.0.0.0` sends the ping to `127.0.0.1`, which is compatible with my expectations, given that a service bound to `0.0.0.0` will also be available at `127.0.0.1`.

Perhaps we should default to `127.0.0.1` instead then? Or is there an RFC somewhere stating what `0.0.0.0` means from a client perspective?

Reply:
Whoa, thank you for your comment. I must have been thinking about the fact that the collector healthcheck defaults to listening to 0.0.0.0:13133.  But that's totally irrelevant to the probe configuration.  Clearly I should have had some more coffee before looking at this one. 🤦‍♂️

The probe should not be configured with a host, it should just be the port and the path. Per `corev1.HTTPGetAction`, 
```golang
// Host name to connect to, defaults to the pod IP. You probably want to set "Host" in httpHeaders instead.
```

I'll get that fixed up.